File: design/kotlin/housepour/HPPill.kt
package ca.lucianlabs.housepour

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

enum class HPPillTone { NEUTRAL, GOLD, BAD }

/** `neutral`: muted on bg2 / line2 border. `gold`: accent on accentSoft / accent 35%. `bad`: bad on bad 6% / bad 45%. */
@Composable
fun HPPill(text: String, tone: HPPillTone = HPPillTone.NEUTRAL, modifier: Modifier = Modifier) {
    val shape = RoundedCornerShape(HPTokens.Radius.pill)
    val (fill, border, color) = when (tone) {
        HPPillTone.NEUTRAL -> Triple(HPTokens.Colors.bg2, HPTokens.Colors.line2, HPTokens.Colors.muted)
        HPPillTone.GOLD -> Triple(HPTokens.Colors.accentSoft, HPTokens.Colors.accent.copy(alpha = 0.35f), HPTokens.Colors.accent)
        HPPillTone.BAD -> Triple(HPTokens.Colors.bad.copy(alpha = 0.06f), HPTokens.Colors.bad.copy(alpha = 0.45f), HPTokens.Colors.bad)
    }
    HPText(
        text = text,
        style = HPTokens.Type.pill,
        color = color,
        modifier = modifier
            .background(fill, shape)
            .border(HPTokens.BORDER_WIDTH.dp, border, shape)
            .padding(horizontal = HPTokens.Space.pillX, vertical = HPTokens.Space.pillY),
        maxLines = 1,
    )
}

File: design/swift/HousePour/HPPill.swift
// House Pour kit — HPPill (COMPONENTS.md "Controls").

import SwiftUI

public enum HPPillTone: Equatable { case neutral, gold, bad }

public struct HPPill: View {
    let text: String
    let tone: HPPillTone
    public init(_ text: String, tone: HPPillTone = .neutral) { self.text = text; self.tone = tone }

    private var fill: Color {
        switch tone {
        case .neutral: return HPTokens.Colors.bg2
        case .gold: return HPTokens.Colors.accentSoft
        case .bad: return HPTokens.Colors.bad.opacity(HPAlpha.badPillFill)
        }
    }
    private var line: Color {
        switch tone {
        case .neutral: return HPTokens.Colors.line2
        case .gold: return HPTokens.Colors.accent.opacity(HPAlpha.goldPillLine)
        case .bad: return HPTokens.Colors.bad.opacity(HPAlpha.badPillLine)
        }
    }
    private var color: Color {
        switch tone {
        case .neutral: return HPTokens.Colors.muted
        case .gold: return HPTokens.Colors.accent
        case .bad: return HPTokens.Colors.bad
        }
    }

    public var body: some View {
        let shape = Capsule(style: .continuous)
        Text(text)
            .hpStyle(HPType.pill, color: color)
            .lineLimit(1)
            .padding(.vertical, HPTokens.Space.pillY)
            .padding(.horizontal, HPTokens.Space.pillX)
            .background(shape.fill(fill))
            .hpBorder(shape, color: line)
            .animation(HPMotion.color, value: tone)
    }
}
